### PR TITLE
add NavItem component

### PR DESCRIPTION
**Pull request by GitJustinLau**
## Description
This is the navigation item component. Will be used to assemble the navbar

2 Variants:
type = "single" | "group"

## Link to JIRA ticket
[Include the link here.](https://implentio.atlassian.net/jira/software/projects/IUT/boards/10?selectedIssue=IUT-27)

## Testing Instructions
Outline the steps required to test your changes effectively (Replace the following to what makes sense for your PR):
- Fetch and checkout my branch `git fetch && git checkout <feat/IUT-27/NavItem>`
- npm run storybook
- Check to see that the elements are aligned, text is properly showing, and there is a hover effect changing text color to brand-emphasis

## Checklist:
Check what is applicable by putting an 'x' in between the []:
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have tested my code thoroughly.
- [ ] I made documentation changes (optional).

### CODE REVIEW

File: src/components/NavItem/NavItem.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+import * as HeroIcons from "@heroicons/react/24/outline";
+import { ChevronLeftIcon } from "@heroicons/react/24/outline";
+
+export type NavItemProps = React.HTMLAttributes<HTMLDivElement> & {
+  type: "single" | "group";
+  title: string;
```
Comment (joseph-hc-jun) [inline src/components/NavItem/NavItem.tsx:7]: Maybe we need another boolean prop named `active` for a different style when the item is selected.

**Comment by joseph-hc-jun** [comment]
Thanks for the quick PR!. I'd like to discuss with you on the approach here for the possibility to use `Accordion` component in shadcn/ui.

File: src/components/NavItem/NavItem.tsx (hunk 1 above)
Comment (GitJustinLau) [reply src/components/NavItem/NavItem.tsx:7]: ok!

File: src/components/NavItem/NavItem.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      {...props}
+    >
+      <div className="flex gap-2 justify-center items-center group-hover:text-stroke-brand-emphasis">
+        <NavItemIcon className="w-5 h-5" />
+        <span>{title}</span>
+      </div>
+      {type === "group" && (
+        <ChevronLeftIcon className="w-4 h4 group-hover:text-stroke-brand-emphasis" />
+      )}
+    </div>
```
Comment (neli-rashkova) [inline src/components/NavItem/NavItem.tsx:31]: `className` is a prop that seems to not be used in your component. If its not necessary, you can remove it.

File: src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import * as React from "react"
```
Comment (joseph-hc-jun) [inline src/components/Tooltip/Tooltip.tsx:1]: I'm wondering why these files are shown as your changes, do you remember whether you used `git merge main` or `rebase`?

Comment (GitJustinLau) [reply src/components/Tooltip/Tooltip.tsx]: oh there was a rebase issue. now fixed

File: src/components/NavItem/NavItem.tsx (hunk 2 above)
Comment (GitJustinLau) [reply src/components/NavItem/NavItem.tsx]: added. i think it helps when users want to add additional classnames

File: src/components/NavItem/NavItem.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  icon,
+  alerts,
+  hotkey,
+  active,
+  ...props
+}: NavItemProps) => {
+  const NavItemIcon = HeroIcons[icon];
+  return (
+    <div
+      className={"flex items-center px-2 py-3 w-[208px]  font-medium group "+`${type==="single" && "data-[active=true]:bg-bg-color-brand-disabled"}` + className} data-active={active}
```
Comment (joseph-hc-jun) [inline src/components/NavItem/NavItem.tsx:26]: Here as well, let's let the parent component dictate the width by giving it 100% width here.

Comment (joseph-hc-jun) [inline src/components/NavItem/NavItem.tsx:26]: Just a note for now, but once the `text-` style conflicts get resolved, let's change `font-medium` to the named variable instead.

Comment (joseph-hc-jun) [inline src/components/NavItem/NavItem.tsx:26]: Question: is `data-active={active}` necessary for giving the right style on active state? Or is it for something else?

**Review by joseph-hc-jun — CHANGES_REQUESTED** [review]
We're almost there I feel, left a bit more comments including a question for you!

File: src/components/NavItem/NavItem.tsx (hunk 4 above)
Comment (GitJustinLau) [reply src/components/NavItem/NavItem.tsx:26]: sounds good. once it is resolved i will review my components

Comment (GitJustinLau) [reply src/components/NavItem/NavItem.tsx:26]: it needs to be passed into the div so that the tailwind class can pick up the state of data-active

Comment (GitJustinLau) [reply src/components/NavItem/NavItem.tsx:26]: yep. you're right. ill be more careful to apply changes to all my PRs for any changes that can apply to more than one

**Review by joseph-hc-jun — APPROVED** [review]

**Pull request merged**